src/Resolver.js

```diff
@@ -159,7 +159,8 @@ export default class Resolver extends React.Component {
         const factory = resolve[name];
         const value = factory(props);
 
-        if (value instanceof Promise) {
+        // duckpunched promise
+        if (value instanceof Promise || typeof value.then === 'function') {
```

Comment:
eslint was eating it for me. ran `eslint src` (instead of `npm run lint`) to get the error. All updated now.